### PR TITLE
Added a docker file to install Parsl

### LINKED ISSUE #6: Docker image for Parsl
Creating a new subtask by breaking down from #2.
Docker image with the make-check running.

### PR DESCRIPTION
**Pull request by hategan**
Fixes #6

### CODE REVIEW

**Review by andre-merzky — APPROVED** [review]
LGTM!

File: docker/Parsl/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+# Recent versions of cryptography require Rust (the language) to be
+# installed. However, the version that comes with Centos 7 is too old.
+# Installing Rust using the install script on rust-lang.org requires
+# interactivity.
+# cyptography 3.3.2 appears to be a recent version that does not use
+# the rust build, so we'll use that until a more sensible solution
+# is found.
+
+run python3 -m pip install "cryptography==3.3.2" parsl
```
Comment (SteVwonder) [inline docker/Parsl/Dockerfile:17]: I think this version pinning solution is totally fine.  Down the road if you want to use a newer version of cryptography, you can install rust via rustup in an unattended fashion by passing `-s -- -y` to the shell that the rustup script is piped to: https://stackoverflow.com/a/57251636/672128

**Review by SteVwonder — APPROVED** [review]
LGTM!

**Comment by dongahn** [comment]
This received two approvals and should go in. I believe this should be merged cleanly. I will click the button.

**Pull request merged**